qiskit_nature/properties/__init__.py

```diff
@@ -0,0 +1,43 @@
+# This code is part of Qiskit.
```

Comment:
The idea which I was following was that the properties are "quantization-agnostic" if you will. I.e. the same kind of properties should be usable in second- and first-quantization problems (once we have those). This will obviously require a `first_q_ops` addition to the interface (and possibly require different raw storage format) but I think in terms of usability this will simplify things for the user.

Reply:
I don;t know - I guess it depends on how these are exposed to the stack the user is dealing with and how much of them they ever use. Given drivers before are quantization dependent and the problem they will be used to solve - its hard for me to see what the simplification truly becomes for the user.

Reply:
I think we have to options here:

#### qiskit_nature.properties
In this case we do **not** add submodules for different problem types and instead expose different operator converters on each property:
```
Property.second_q_ops()
Property.first_q_ops()
Property.sampling_q_ops()
etc...
```
If a property does not support a specific kind of problem, we can raise an error which should gracefully be handled by the calling frame.

The benefit would be a single property implementation with the potential to avoid code duplication and the ability to reuse the same internal raw data for different kind of problems.
However, the latter might also be a disadvantage if the raw formats for different problems for the same property are too diverse. In that case, we may be better off with the following option.

#### qiskit_nature.properties.{second_quantization,first_quantization,sampling,...}
This mimics the rest of stack layout but may lead to code duplication. Arguably, we should then have something like:
```
SecondQuantizedProperty.operators()
FirstQuantizedProperty.operators()
...
```

For now, I would prefer to go with option one. I think only time will tell whether we really need the second case.

Reply:
I just wonder as this ends up giving two different patterns in regards of quantization. A property behaving as a Union with different methods for different quantizations, but other aspects being split into different types e.g. ElectronicStructureProblem has a `second_q_ops` - on the same basis one might expect it could have been similar and have a `first_q_ops` as well.

Consistency is also an aspect that helps people with the overall code - properties this way will be different than all aspects of the stack around it. Maybe a Union is simpler in that I only have one class to go to - but now offset that against it being different in the way quantization is dealt with.

Reply:
I have relocated the code as per your suggestion. I agree that the consistency argument outweighs the others :+1: